src/MemoryRouter.test.tsx

```diff
@@ -417,6 +417,55 @@ describe("MemoryRouter", () => {
           });
         });
 
+        describe("search parameter", () => {
+          it("happy path", async () => {
+            await memoryRouter.push({
+              pathname: "/path",
+              search: "foo=FOO&bar=BAR",
+            });
+            expectMatch(memoryRouter, {
+              asPath: "/path?foo=FOO&bar=BAR",
+              pathname: "/path",
+              query: {
+                foo: "FOO",
+                bar: "BAR",
+              },
+            });
+          });
+
+          it("multiple values can be specified for a query parameter", async () => {
+            await memoryRouter.push({
+              pathname: "/path",
+              search: "foo=FOO&foo=BAR",
+            });
+            expectMatch(memoryRouter, {
+              asPath: "/path?foo=FOO&foo=BAR",
+              pathname: "/path",
+              query: {
+                foo: ["FOO", "BAR"],
+              },
+            });
+          });
+
+          it("if search and query are both provided preference is given to search", async () => {
```

Comment:
Not a big deal, but did you verify that this is how Next behaves?  Gives preference to `search` over `query`?

Reply:
yes exactly, tested this manually and this seems to be the behaviour